packages/@tinacms/cli/src/cmds/waitForDB/index.ts

```
@@ -0,0 +1,109 @@
+const STATUS_FAILED = 'failed'
+
+interface IndexStatusResponse {
+  status: 'inprogress' | 'complete' | 'failed' | 'unknown'
+  timestamp: number
+  error?: string
+}
+
+export const waitForDB = async (ctx, next, options: { verbose?: boolean }) => {
+  if (options.verbose) {
+    logger.info(logText('Waiting for DB...'))
+  }
+
+  if (!ctx.client) {
+    if (options.verbose) {
+      logger.info(logText('client is unavailable, skipping...'))
+    }
+    return next()
+  }
+
+  const client = ctx.client
+
+  const { host, clientId, branch, isLocalClient } = client.parseURL()
+
+  if (isLocalClient) {
+    if (options.verbose) {
+      logger.info(logText('client is local, skipping...'))
+    }
+    return next()
+  }
+
+  const pollForStatus = async () => {
+    try {
+      if (options.verbose) {
+        logger.info(logText('Polling for status...'))
+      }
+      const headers = new Headers()
+      headers.append('Content-Type', 'application/json')
+      if (client.readonlyToken) {
+        headers.append('X-API-KEY', client.readonlyToken)
+      }
+
+      const response = await fetch(
+        `https://${host}/db/${clientId}/status/${branch}`,
+        {
+          method: 'GET',
+          headers,
+        }
+      )
+      const { status, error } = (await response.json()) as IndexStatusResponse
+
+      const statusMessage = `DB responded with: '${status}'`
+
+      // Index Complete
+      if (status === STATUS_COMPLETE) {
+        if (options.verbose) {
+          logger.info(logText(`${statusMessage}`))
+        }
+        return next()
+
+        // Index Inprogress
+      } else if (status === STATUS_INPROGRESS) {
+        if (options.verbose) {
+          logger.info(logText(`${statusMessage}, trying again in 5 seconds`))
+        }
+        setTimeout(pollForStatus, POLLING_INTERVAL)
+
+        // Index Failed
+      } else if (status === STATUS_FAILED) {
+        logger.error(
+          dangerText(`${statusMessage}, encountered error: ${error}`)
```

We should probably throw an error here (With an message) instead of doing a logger.error. Failing silently causes a very strange and cryptic  error message on vercel.